sites/all/modules/custom/jdt_user_notes/jdt_user_notes.module

```diff
@@ -5,25 +5,367 @@
  */
 
 /**
- * Implements hook_permission() for jdt_user_notes().
- *
- * Controls permissions.
+ * Include CRUD file.
+ */
+include_once 'jdt_user_notes.crud.inc';
+
+/**
+ * Implements hook_permission().
  */
 function jdt_user_notes_permission() {
   $permissions = array(
-    'creating notes' => array(
-      'title' => t('Add a Note'),
+    'read own notes' => array(
+      'title' => t('Read Own Notes'),
+    ),
+    'read any notes' => array(
+      'title' => t('Read Any Notes'),
+    ),
+    'create new note' => array(
+      'title' => t('Create a Note'),
+    ),
+    'edit own note' => array(
+      'title' => t('Edit Own Note'),
     ),
-    'reading notes' => array(
-      'title' => t('Read a Note'),
+    'edit any note' => array(
+      'title' => t('Edit Any Note'),
     ),
-    'updating notes' => array(
-      'title' => t('Edit a Note'),
+    'delete own note' => array(
+      'title' => t('Delete Own Note'),
     ),
-    'deleting notes' => array(
-      'title' => t('Delete a Note'),
+    'delete any note' => array(
+      'title' => t('Delete Any Note'),
     ),
   );
 
   return $permissions;
 }
+
+/**
+ * Implements hook_menu().
+ */
+function jdt_user_notes_menu() {
+  $items = array();
+
+  $items['node/%node/add/%ctools_js'] = array(
+    'title' => 'Add a Note',
+    'page callback' => 'jdt_user_notes_add',
+    'page arguments' => array(1, 3),
+    'access arguments' => array('create new note'),
+  );
+
+  $items['node/%node/edit/%/%ctools_js'] = array(
+    'title' => 'Edit a Note',
+    'page callback' => 'jdt_user_notes_edit',
+    'page arguments' => array(1, 3, 4),
+    'access arguments' => array('edit own note'),
+  );
+
+  $items['node/%node/notes'] = array(
+    'title' => 'My Notes',
+    'page callback' => 'jdt_user_notes_display',
+    'page arguments' => array(1),
+    'access callback' => 'jdt_user_notes_access',
+    'access arguments' => array(1),
```

Comment:
you can add another access argument (like which permission) so it's passed to the function `jdt_user_notes_access` - so that this function could be re-used. Just a suggestion